### PR TITLE
WFLY-11698 fixed max-delivery-attempts settings by TransactionTimeout…

### PR DESCRIPTION
**Pull request by istudens**
…QueueSetupTask

https://issues.jboss.org/browse/WFLY-11698

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java [hunk 1]
```diff
@@ -83,21 +83,25 @@ public void tearDown(ManagementClient managementClient, String containerId) thro
 
     // /subsystem=messaging-activemq/server=default/address-setting=queue.propertyTimeoutQueue:add(max-delivery-attempts=1)
```
Comment (ochaloup) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java]: would you be so kind and change (or remove) the comment to reflect the changes in the method (aka. `address-setting=queue.` should be `address-setting=jms.queue.`)

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java [hunk 2]
```diff
@@ -83,21 +83,25 @@ public void tearDown(ManagementClient managementClient, String containerId) thro
...
-            .get(ClientConstants.OP).set(ClientConstants.ADD)
-            .get("max-delivery-attempts").set(1);
+        ModelNode address = amqServerDefaultAdress.clone().add("address-setting", "jms.queue." + queueName);
+
+        ModelNode operation = new ModelNode();
+        operation.get(ClientConstants.OP).set(ClientConstants.ADD);
+        operation.get(ClientConstants.OP_ADDR).set(address);
+        operation.get("max-delivery-attempts").set(1);
+        // delay redelivery of messages for 5s
+        operation.get("redelivery-delay-multiplier").set(1.5);
```
Comment (ochaloup) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/TransactionTimeoutQueueSetupTask.java]: thank you for correcting me in wrongly put fluent calls of the `ModelNode().get(...).set(..)...` which do not work!

from my testing I think the `max-delivery-attempts` could be enough. the others do not add much different behaviour.

**Review by ochaloup — COMMENTED** [review]
@istudens thanks for fixing my omissions :sweat_smile:. I added just cosmetic comments to this.

**Comment by istudens** [comment]
Fixed, thanks.
Just a note that I omitted all the changes around SystemException handling as it looks it can't be hit after WFTC-58 change anymore.

**Review by ochaloup — APPROVED** [review]
@istudens +1, yeap, as we discussed, I know about it. Thank you!

**Comment by ochaloup** [comment]
@istudens one more thing, can you add one more change to your PR, please? It's what @bstansberry  asked at my original PR (https://github.com/jbosstm/narayana/pull/1403) which is make the `ModelNode amqServerDefaultAdress` `private`?

**Comment by istudens** [comment]
@ochaloup done

**Review by bstansberry — APPROVED** [review]

**Pull request merged**